src/main/java/net/minecraftforge/server/command/CommandTreeBase.java

```diff
@@ -72,19 +101,28 @@ public ICommand getSubCommand(String command)
 
     public Map<String, ICommand> getCommandMap()
     {
-        return Collections.unmodifiableMap(commandMap);
+        return commandMap;
     }
 
     public List<ICommand> getSortedCommandList()
     {
-        List<ICommand> list = new ArrayList<>(getSubCommands());
-        Collections.sort(list);
-        return list;
+        if (sortedCommands == null)
+        {
+            sortedCommands = new ArrayList<>(commandMap.values());
+            sortedCommands.sort(null);
+        }
+
+        return sortedCommands;
+    }
+
+    public void updateSortedCommandList()
```

Comment:
I don't like this name; you're not really updating it, just invalidating it.

Reply:
refreshSortedCommandList() or clearSortedCommandListCache() maybe?

Reply:
clear/invalidate cache would work